projects/common/src/time/time-range.service.test.ts

```diff
@@ -147,6 +147,44 @@ describe('Time range(TR) service', () => {
     });
   });
 
+  test('Skips invalid URL time range and emits default TR when set', () => {
+    const invalidTimeRange = new FixedTimeRange(new Date(1573277100277), new Date(1573277100277));
+    const defaultTimeRange = new FixedTimeRange(new Date(1573277100277), new Date(1573277100999));
+    const mockNavigation$ = new Subject();
+    runFakeRxjs(({ expectObservable, cold }) => {
+      const spectator = buildService({
+        providers: [
+          mockProvider(NavigationService, {
+            navigation$: mockNavigation$.asObservable().pipe(
+              map(
+                timeRangeString =>
+                  (({
+                    queryParamMap: of(convertToParamMap({ time: timeRangeString, refresh: 'true' })),
+                    snapshot: {
+                      queryParamMap: convertToParamMap({ time: timeRangeString, refresh: 'true' })
+                    }
+                  } as unknown) as ActivatedRoute)
+              )
+            ),
+            addQueryParametersToUrl: (newParams: QueryParamObject) => mockNavigation$.next(newParams.time as string),
+            getQueryParameter: jest.fn().mockReturnValueOnce('').mockReturnValueOnce(defaultTimeRange.toUrlString()),
+            getCurrentActivatedRoute: () =>
+              (({ snapshot: { queryParams: { time: 'test-value' } } } as unknown) as ActivatedRoute),
+            replaceQueryParametersInUrl: jest.fn()
+          })
+        ]
+      });
+
+      cold('u').subscribe(() => mockNavigation$.next(invalidTimeRange.toUrlString()));
```

Comment:
For the timing of the observables i thought, but i guess since we're only expecting a single emission we shouldn't need to use it ? i'll try

Reply:
in the earlier one it was used for delaying calls and seeing how late arriving things show up in the time range observable, but here we don't need that, since we're just checking the default